### PR TITLE
feat(scopebody): Extract scopes from body in mock

### LINKED ISSUE #153: Mock tooling: Extract scopes from body
See https://github.com/zalando-incubator/authmosphere/blob/master/src/mock-tooling/index.ts#L45

### PR DESCRIPTION
**Pull request by Retro64**
close #153

### CODE REVIEW

File: test/unit/utils.ts [hunk 1]
```diff
@@ -113,16 +113,16 @@ describe('utils', () => {
 
       return expect(result).to.equal(true);
     });
-  });
 
-  it('should return false, if credentialsDir do not exists in config', () => {
-    const config = {
-      credentialsDir: undefined
-    };
+    it('should return false, if credentialsDir do not exists in config', () => {
```
Comment (ISO50) [inline test/unit/utils.ts]: What exactly changed here? Looks identically

File: src/utils.ts [hunk 2]
```diff
@@ -13,6 +13,14 @@ import {
    CredentialsUserClientConfig
 } from './types';
 
+/**
+ * Converts an form-urlencoded string to a JSON formatted string
+ *
+ * @param requestBody form-urlencoded string
+ */
+const formURLencodedToJSONformatted = (requestBody: string) =>
```
Comment (bzums) [inline src/utils.ts]: Suggestion for a little bit increased readability:

```ts
const formURLencodedToJSONformatted = (requestBody: string) =>
  `{${
      requestBody
        .replace(RegExp('&', 'g'), '","')
        .replace(RegExp('=', 'g'), '":"')
    }}`;
```

File: test/unit/utils.ts [hunk 3]
```diff
@@ -136,113 +136,135 @@ describe('utils', () => {
...
-      grantType: OAuthGrantType.CLIENT_CREDENTIALS_GRANT,
-      applicationUsername: 'applicationUsername',
-      applicationPassword: 'applicationPassword',
-      clientId: 'clientId',
-      clientSecret: 'clientSecret'
-    };
+  describe('formURLencodedToJSONformatted', () => {
+    it('should converts an form-urlencoded string to a JSON formatted string', () => {
+      const config = {
+        ...{},
```
Comment (bzums) [inline test/unit/utils.ts]: why use spread here at all?

**Comment by bzums** [comment]
👍

**Comment by Retro64** [comment]
:+1:

**Pull request merged**